addons/api/addon/serializers/host-catalog.js

```diff
@@ -68,15 +68,36 @@ export default class HostCatalogSerializer extends ApplicationSerializer {
     const value = super.serializeAttribute(...arguments);
     const { compositeType, credentialType, isPlugin } = snapshot.record;
     const { options } = attribute;
-
     const fields =
       compositeType === TYPE_HOST_CATALOG_PLUGIN_AZURE
         ? azureFields
         : AWSfieldsWithCredentialType[credentialType];
 
-    // Delete any nested attribute fields that don't belong to the record type
+    // Delete any nested attribute fields that don't belong to the record's compositeType
     if (isPlugin && options.isNestedAttribute && json.attributes) {
-      if (!fields.includes(key)) {
+      // If compositeType is AWS, delete azure fields from the json
+      if (compositeType === TYPE_HOST_CATALOG_PLUGIN_AWS) {
+        if (
+          !AWSfieldsWithCredentialType[credentialType].includes(key) &&
+          azureFields.includes(key)
+        ) {
+          delete json.attributes[key];
+        }
+
+        // If compositeType is AWS, nullify AWS fields that don't belong to the credentialType(static-credential or dynamic-credential)
+        if (
+          !AWSfieldsWithCredentialType[credentialType].includes(key) &&
+          !azureFields.includes(key)
+        ) {
+          json.attributes[key] = null;
+        }
+      }
+
+      // If compositeType is Azure, delete AWS fields from the json
+      if (
+        compositeType === TYPE_HOST_CATALOG_PLUGIN_AZURE &&
+        !azureFields.includes(key)
+      ) {
```

Comment:
cleaner way would be to introduce a  new `compositeType` field in model `@atrr` 

if you are ok with that idea, I can refactor that in a separate PR. Let me know your thoughts on this approach

Reply:
Not sure i'm following along what introducing new `compositeType` entails. Does this mean that instead of `compositeType` being just `aws`  it would change to be something more specific depending on credential type ?

Reply:
yeah, right now we pass `for` which is the recordType. So if we could pass `compositeType` or `credentialType` from the model it would be a lot cleaner compared to how we handle this in a resource serializer

Reply:
ah okay that makes sense, sounds like a good idea and would help with readability